apps/dynamo/lib/dynamo/vnode.ex

```diff
@@ -181,10 +185,21 @@ defmodule Vnode do
       |> Map.get(index, %{})
       |> Map.get(key, :key_not_found)
 
-    send(sender, {:ok, key, value, state.partition})
+    send(sender, {:ok, key, value, state.partition, Node.self()})
     {:noreply, state}
   end
 
+  @impl true
+  def handle_cast({:update_repl, key, value, index, context}, state) do
```

Comment:
Seems like `def handle_cast({:repl, sender, index, key, value, context}, state) do` not send back a reply. Why don't you modify the `:repl` command to have flag to send back the reponse or not?